submissions/mustbefail/dom-api-task/src/js/app.js

```diff
@@ -0,0 +1,54 @@
+import makeModal from './modal.js';
+
+const makeMsg = (ticketsAmount, totalPrice) => {
+  if (!ticketsAmount) return 'No seat selected';
+  return `You have selected ${ticketsAmount} seats for a price of $${totalPrice}`;
+};
+
+const render = ({ ticketsAmount, totalPrice }) => {
+  const cartTip = document.querySelector('.add-to-cart-block p');
+  cartTip.textContent = makeMsg(ticketsAmount, totalPrice);
+};
+
+export default () => {
+  const state = {
+    ticketPrice: 24,
+    cart: [],
+    totalPrice: null,
+    ticketsAmount: null,
+  };
+
+  const form = document.querySelector('form');
+  form.addEventListener('click', (event) => {
+    const { currentTarget } = event;
+
+    const seats = currentTarget.querySelectorAll('input:checked');
+    const seatsData = Array.from(seats).map((seat) => seat.value);
```

Comment:
Using `Array.from` on Node List, not a good idea. NodeList already has a `forEach` use him.
Avoid this pattern or change logic.)

Reply:
in this case, I use array conversion to work with data, not to search for data. I can replace Array.from with a spred operator (which is the same). If I use forEach, I need to add each value to state.cart and clear this field after rendering. In any case, I need an array of data, so I chose this option.